crates/signer/src/manager/dirk.rs

```diff
@@ -0,0 +1,917 @@
+        for host in config.hosts.clone() {
+            let mut tls_config = ClientTlsConfig::new().identity(config.client_cert.clone());
+
+            if let Some(ca) = &config.cert_auth {
+                tls_config = tls_config.ca_certificate(ca.clone());
+            } else {
+                trace!(?host.domain, "CA certificate for domain not found");
+            }
+
+            tls_config = tls_config.domain_name(host.domain.unwrap_or_default());
+            tls_configs.push(tls_config);
+        }
+
+        // Create a channel for each host, attempt to connect,
+        // and save it into a hashmap Host->Channel
+        let mut channels = HashMap::new();
+        for (i, tls_config) in tls_configs.iter().enumerate() {
+            let config_host = config.hosts[i].clone();
+            let domain = config_host.domain.unwrap_or_default();
+            match Channel::from_shared(config_host.url.to_string())
+                .map_err(|_| eyre::eyre!("Invalid Dirk URL"))?
+                .tls_config(tls_config.clone())
+                .map_err(|_| eyre::eyre!("Invalid Dirk TLS config"))?
+                .connect()
+                .await
+            {
+                Ok(ch) => {
+                    channels.insert(domain, ch);
+                }
+                Err(e) => {
+                    trace!("Couldn't connect to Dirk with domain {}: {e}", domain);
+                }
+            }
+        }
+
+        let mut accounts: HashMap<String, Account> = HashMap::new();
+
+        for host in config.hosts {
+            let domain = host.domain.unwrap_or_default();
+            let channel = channels
+                .get(&domain)
+                .ok_or(eyre::eyre!("Couldn't connect to Dirk with domain {domain}"))?
+                .clone();
+
+            let (dirk_accounts, dirk_distributed_accounts) = get_accounts_in_wallets(
+                channel.clone(),
+                host.accounts
+                    .iter()
+                    .filter_map(|account| Some(account.split_once("/")?.0.to_string()))
+                    .collect(),
+            )
+            .await?;
+
+            for account_name in host.accounts.clone() {
+                let (wallet, _) = account_name.split_once("/").ok_or(eyre::eyre!(
+                    "Invalid account name: {account_name}. It must be in format wallet/account"
+                ))?;
+
+                // Handle simple accounts
+                for dirk_account in dirk_accounts.iter().filter(|a| {
+                    a.name == account_name || a.name.starts_with(&format!("{}/", account_name))
+                }) {
+                    let public_key = BlsPublicKey::try_from(dirk_account.public_key.as_slice())?;
+                    let key_name =
+                        dirk_account.name.split_once("/").map(|(_, n)| n).unwrap_or_default();
+                    trace!(?dirk_account.name, "Adding account to hashmap");
+                    let is_proxy = key_name.contains('/');
```

Comment:
Would be nice to check if it has the exact format `<consensus>/<module_id>/<uuid>`

Reply:
Should be done in 51fe16dcb1b307f9a0b6450ad3ad3ee62a7cfcda but I want to test it a bit more just in case.